### ISSUE TITLE
Stabilize a hingeConstraint

### ISSUE THREAD

**Issue by tomo0613** [schteppe/cannon.js#286]
Greetings!

Is there a way to make a hingeConstraint stable (stiff) ?
I'm trying to make a rotor pivot for my helicopter. Actually it works, but acts like as it would attached with a rubber string. If u can check the link under, and u start just pressing A or D, than you can see a weird thing.
My goal is to make the rotor, only be able to rotate around the Y axis, even if it's colliding with something.
https://cdn.rawgit.com/tomo0613/test-project-repo/master/index.html

I checked this issue [ https://github.com/schteppe/cannon.js/issues/63 ]  as well but it didn't help.

**Comment by schteppe** [comment]
Hi!
The easiest way to fix this is to set the mass of the rotor to something bigger. Other than that you can try setting relaxation and stiffness on the hingeConstraint.

```
var stiffness = 1e10;
var relaxation = 4;
var timeStep = 1/60;
hinge.rotationalEquation1.setSpookParams(stiffness,relaxation,timeStep);
hinge.rotationalEquation2.setSpookParams(stiffness,relaxation,timeStep);
```

Not sure if this code will run in your slightly older version of cannon, but yea, should work on master branch.

Another question... Why add the rotor to the physics? At higher rotor speeds you will be forced to use a super small timeStep, to make the rotor collide at all. And a super small physics timeStep will kill performance. Just saying :) Alternatively you could have a cylinder block for the rotor sweep, that is a shape inside the helicopter body.

**Comment by tomo0613** [comment]
That make sense. (your question)
Thank u for your advice!
I played with the mass, and just made it even worse, and at the moment that ...setSpookParams();
doesn't want to work for me.
But all together, as u wrote, looks like better if i look for a different solution.
Actually I just want to detect the collision for the body and the rotor separately.

**Comment by schteppe** [comment]
All right, cool. Then you should be able to use the shape collision events. Same thing here, you need to be on latest master branch, probably.

``` js
world.addEventListener('beginShapeContact', function (evt) {
    if(evt.shapeA === rotorShape || evt.shapeB === rotorShape){
        console.log('rotorShape in contact');
    } else if(evt.shapeA === helicopterShape || evt.shapeB === helicopterShape){
        console.log('helicopterShape in contact');
    }
});
world.addEventListener('endShapeContact', function(evt){
    // ...do similar stuff...
});
```

**Comment by tomo0613** [comment]
Great! I'll try that. (actually now I already did it :) )
So:
this link [ https://cdn.rawgit.com/schteppe/cannon.js/master/build/cannon.min.js ] should provide the latest version of CannonJS if I'm right.
I just modified the code, but those shape collision events doesn't want to work.
I can't find any reference or demo to see how should  i use them.
Probably should work with compound shapes, but i cant find out how.
Can u give me a hint?
[ https://cdn.rawgit.com/tomo0613/test-project-repo/master/index.html ]
- my previous idea before i wrote the above, was: to create a static cylinder for the rotor (like now just separately, outside from the compound), and attach to the compound with a lockConstraint or contactEquation. The results was pretty similar to the rubber like hingeConstraint solution.
  Are these constraints for different purposes or i just messed something?
